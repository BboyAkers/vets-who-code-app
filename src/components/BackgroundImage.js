import React from 'react'
import { graphql, StaticQuery } from 'gatsby'
import PropTypes from 'prop-types'
import BackgroundImage from 'gatsby-background-image'

const BackgroundSection = ({ fileName, className, children }) => (
  <StaticQuery
    query={graphql`
      query {
        allImageSharp {
          edges {
            node {
              fluid(quality: 90, maxWidth: 4100) {
                ...GatsbyImageSharpFluid_withWebp
                originalName
              }
            }
          }
        }
      }
    `}
    render={data => {
      // Set ImageData.
      const image = data.allImageSharp.edges.find(edge => edge.node.fluid.originalName === fileName)
      if (!image) {
        return null
      }
      return (
        <BackgroundImage
          Tag="section"
          className={className}
          fluid={image.node.fluid}
          style={{ height: '80vh' }}
        >
          {children}
        </BackgroundImage>
      )
    }}
  />
)

BackgroundSection.propTypes = {
  fileName: PropTypes.string.isRequired,
  className: PropTypes.string,
  children: PropTypes.node,
}

export default BackgroundSection
