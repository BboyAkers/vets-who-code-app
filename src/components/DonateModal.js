import React from 'react';
import Modal from 'react-modal';

Modal.setAppElement('#___gatsby');

const container = {
  content: {
    top: '20%',
    bottom: '20%',
    height: 'auto'
  }
};

const closeButton = {
  right: '8px'
};

export default function DonateModal({ modalIsOpen, openModal, closeModal }) {
  return (
    <Modal
      style={container}
      contentLabel="Donate"
      isOpen={modalIsOpen}
      onRequestClose={closeModal}
      shouldCloseOnOverlayClick={true}
    >
      <iframe
        src="https://donorbox.org/embed/vetswhocode-donation?show_content=true"
        style={{
          maxWidth: '100%',
          minWidth: '100%',
          maxHeight: 'none!important'
        }}
        seamless="seamless"
        name="donorbox"
        scrolling="no"
        allowpaymentrequest
        width="100%"
        height="685px"
        frameBorder={0}
      />
    </Modal>
  );
}
