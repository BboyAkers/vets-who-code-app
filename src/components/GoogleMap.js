import React, { useEffect, useRef } from 'react'

function GoogleMap() {
  const googleMapRef = useRef()

  useEffect(() => {
    let current = true
    if (current) {
      mapInit()
    }
    return () => (current = false)
  })

  function mapInit() {
    const googleMapScript = document.createElement('script')
    googleMapScript.src =
      'https://maps.googleapis.com/maps/api/js?key=AIzaSyBQt6WATWNedQ8TSM7sCKOI1uoPR2JrG-4'
    window.document.body.appendChild(googleMapScript)

    googleMapScript.addEventListener('load', () => {
      createGoogleMap()
      createMarker()
    })
  }

  function createGoogleMap() {
    new window.google.maps.Map(googleMapRef.current, {
      zoom: 16,
      scrollwheel: false,
      center: {
        lat: 36.1579519,
        lng: -86.7708364,
      },
      disableDefaultUI: true,
    })
  }

  function createMarker() {
    const mapCanvas = googleMapRef.current
    const mapOptions = {
      center: new google.maps.LatLng(36.1579519, -86.7708364),
      scrollwheel: false,
      zoom: 16,
      mapTypeId: google.maps.MapTypeId.ROADMAP,
    }
    const map = new google.maps.Map(mapCanvas, mapOptions)
    const contentString = `
      <div id="content">
        <h2>#VetsWhoCode</h2>
        <div id="bodyContent">41 N Peabody st, Nashville Tn, 37120</div>
      </div>
    `
    const myLatLng = { lat: 36.1577981, lng: -86.7707313 }
    const infowindow = new google.maps.InfoWindow({
      content: contentString,
    })

    const marker = new google.maps.Marker({
      position: myLatLng,
      map,
      title: '#VetsWhoCode',
    })

    marker.addListener('click', () => {
      infowindow.open(map, marker)
    })
  }

  return (
    <div
      id="map-canvas"
      ref={googleMapRef}
      className="map-default-height"
      style={{ width: '100%', height: '300px' }}
    />
  )
}

export default GoogleMap
