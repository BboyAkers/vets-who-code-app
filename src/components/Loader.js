import React, { Component } from 'react';

class Loader extends Component {
  componentDidMount() {
    global.jQuery = require('../../static/vendor/jquery/dist/jquery');
    window.$ = jQuery;
    window.jQuery = jQuery;
    this.handleLoader();
  }

  handleLoader = () => {
    $(window).on('load', () => {
      $('#loader').fadeOut();
      $('#loader-wrapper')
        .delay(100)
        .fadeOut();
    });
  }

  render() {
    return (
      <div id="loader-wrapper">
        <div id="loader" />
      </div>
    );
  }
}

export default Loader;
