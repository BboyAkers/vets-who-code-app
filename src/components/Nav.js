import React, { useState, useRef, useEffect } from 'react'
import Link from 'gatsby-link'
import FluidImage from '../components/FluidImage'

function Nav() {
  const navRef = useRef()
  const [isNavOpen, setIsNavOpen] = useState(false)
  const [opacity, setOpacity] = useState(0.9)

  useEffect(() => {
    let current = true
    if (current) {
      document.addEventListener('scroll', handleScroll)
    }
    return () => {
      current = false
      document.removeEventListener('scroll', handleScroll)
    }
  })

  function handleScroll() {
    const winScroll = document.body.scrollTop || document.documentElement.scrollTop
    if (winScroll === 0) {
      setOpacity(0.9)
    }
    if (winScroll > 0) {
      setOpacity(1)
    }
  }

  return (
    <nav
      ref={navRef}
      id="fixedTopNav"
      className="navbar navbar-fixed-top main-navigation navbar-solid"
      itemScope=""
      itemType="https://schema.org/SiteNavigationElement"
      role="navigation"
      style={{ opacity: opacity }}
    >
      <div className="container">
        <div className="navbar-header">
          <div className="navbar-brand" itemScope="" itemType="https://schema.org/Organization">
            <span className="sr-only">#VetsWhoCode</span>
            <Link to="/">
              <FluidImage
                fileName="hashflag_white.jpg"
                alt="#VetsWhoCode Logo"
                className="logo_holder"
              />
            </Link>
          </div>
          <button
            type="button"
            id="hamburger-1"
            onClick={() => setIsNavOpen(!isNavOpen)}
            className={`navbar-toggle collapsed hamburger ${isNavOpen ? 'is-active' : ''}`}
            aria-expanded={isNavOpen ? 'true' : 'false'}
          >
            <span className="line"></span>
            <span className="line"></span>
            <span className="line"></span>
            <span className="sr-only">#VetsWhoCode</span>
          </button>
        </div>
        <div
          className={`navbar-collapse collapse ${isNavOpen ? 'in' : ''}`}
          id="main-nav-collapse"
          aria-expanded={isNavOpen ? 'true' : 'false'}
        >
          <ul className="nav navbar-nav navbar-right" role="menu">
            <li role="menu-item">
              {' '}
              <Link to="/">
                <span>Home</span>
              </Link>
            </li>
            <li role="menu-item">
              {' '}
              <Link to="/about">
                <span>About</span>
              </Link>
            </li>
            <li role="menu-item">
              {' '}
              <Link to="/syllabus">
                <span>Syllabus</span>
              </Link>
            </li>
            <li role="menu-item">
              {' '}
              <Link to="/mentor">
                <span>Mentor</span>
              </Link>
            </li>
            <li role="menu-item">
              {' '}
              <Link to="/apply">
                <span>Apply</span>
              </Link>
            </li>
            <li role="menu-item">
              {' '}
              <Link to="/donate">
                <span>Donate</span>
              </Link>
            </li>
            <li role="menu-item">
              {' '}
              <Link to="/contact">
                <span>Contact Us</span>
              </Link>
            </li>
            <li role="menu-item">
              {' '}
              <a
                href="https://medium.com/vets-who-code"
                without="true"
                rel="noopener noreferrer"
                target="_blank"
              >
                <span>Blog</span>
              </a>
            </li>
          </ul>
        </div>
      </div>
    </nav>
  )
}

export default Nav
