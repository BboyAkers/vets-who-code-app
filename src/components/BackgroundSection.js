import React from 'react'
import { graphql, StaticQuery } from 'gatsby'
import PropTypes from 'prop-types'
import BackgroundImage from 'gatsby-background-image'

const BackgroundSection = ({ fileName, className, children, ...rest }) => (
  <StaticQuery
    query={graphql`
      query {
        allImageSharp {
          edges {
            node {
              fluid(quality: 90, maxWidth: 2100) {
                ...GatsbyImageSharpFluid_withWebp
                originalName
              }
            }
          }
        }
      }
    `}
    render={data => {
      // Set ImageData.
      const image = data.allImageSharp.edges.find(edge => edge.node.fluid.originalName === fileName)
      if (!image) {
        return null
      }
      return (
        <BackgroundImage
          {...rest}
          Tag="section"
          className={className}
          fluid={image.node.fluid}
          style={{ height: '80vh' }}
        >
          {children}
        </BackgroundImage>
      )
    }}
  />
)

BackgroundSection.propTypes = {
  fileName: PropTypes.string.isRequired,
  className: PropTypes.string,
  children: PropTypes.node,
}

export default BackgroundSection
