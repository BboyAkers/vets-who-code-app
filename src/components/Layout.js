import PropTypes from 'prop-types'
import { ToastContainer } from 'react-toastify'
import ThemeProvider from '../store/ThemeProvider'
import Nav from './Nav'
import Footer from './Footer'
import 'react-toastify/dist/ReactToastify.css'

const TemplateWrapper = ({ children }) => (
  <>
    <ThemeProvider>
      <main className="main_container">
        <ToastContainer />
        <Nav />
        {children}
        <Footer />
      </main>
    </ThemeProvider>
  </>
)

TemplateWrapper.propTypes = {
  children: PropTypes.any,
}

export default TemplateWrapper
