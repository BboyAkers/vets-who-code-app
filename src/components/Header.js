import React, { useRef, Fragment } from 'react'
import Slider from 'react-slick'
import { Link } from 'gatsby'
import BackgroundSection from '../components/BackgroundSection'

const settings = {
  infinite: true,
  speed: 500,
  fade: true,
  lazyLoad: true,
  slidesToShow: 1,
  slidesToScroll: 1,
  autoplay: true,
  autoplaySpeed: 3000,
  nextArrow: <Fragment />, // use Fragment to disable react-slick bd
  prevArrow: <Fragment />,
}

function Header() {
  const sliderRef = useRef()

  return (
    <header id="site-header" className="site-header flexslider classic">
      <div style={{ backgroundColor: 'rgb(9, 31, 64)' }}>
        <Slider {...settings} ref={sliderRef}>
          <BackgroundSection fileName="code.jpg" critical fadeIn>
            <div className="header-classic wrapper-table overlay-01">
              <div className="valign-center">
                <div className="container">
                  <div className="col-md-10 col-md-offset-1">
                    <div className="intro text-left">
                      <h1>Learn</h1>
                      <p className="subtitle">How To Code With Other Veterans.</p>
                      <div className="btn-cal-group">
                        {' '}
                        <Link to="/apply" className="btn btn-charity-default">
                          Apply
                        </Link>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </BackgroundSection>
          <BackgroundSection fileName="speaking.jpg">
            <div className="header-classic  wrapper-table overlay-01">
              <div className="valign-center">
                <div className="container">
                  <div className="col-md-10 col-md-offset-1">
                    <div className="intro text-left">
                      <h1>Sign Up</h1>
                      <p className="subtitle">As A Mentor.</p>
                      <div className="btn-cal-group">
                        {' '}
                        <Link to="/mentor" className="btn btn-charity-default">
                          {' '}
                          Sign Up
                        </Link>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </BackgroundSection>
          <BackgroundSection fileName="this_is_us.jpg">
            <div className="header-classic wrapper-table overlay-01">
              <div className="valign-center">
                <div className="container">
                  <div className="col-md-10 col-md-offset-1">
                    <div className="intro text-right">
                      <h1 className="red">
                        <span className="red">Help Us</span>
                      </h1>
                      <p className="subtitle red">Teach More Veterans How To Code.</p>
                      <div className="btn-cal-group">
                        {' '}
                        <Link to="/donate" className="btn btn-charity-default">
                          Please Donate
                        </Link>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </BackgroundSection>
        </Slider>
        <ol className="flex-control-nav flex-control-paging">
          <li>
            <a className="">1</a>
          </li>
          <li>
            <a className="flex-active">2</a>
          </li>
          <li>
            <a className="">3</a>
          </li>
        </ol>
        <ul className="flex-direction-nav">
          <li className="flex-nav-prev" onClick={() => sliderRef.current.slickNext()}>
            <a className="flex-prev" href="#">
              Previous
            </a>
          </li>
          <li className="flex-nav-next" onClick={() => sliderRef.current.slickPrev()}>
            <a className="flex-next" href="#">
              Next
            </a>
          </li>
        </ul>
      </div>
    </header>
  )
}
export default Header
