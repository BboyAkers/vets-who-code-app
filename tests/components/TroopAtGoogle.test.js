import React from 'react'
import { TroopsAtGoogleImage  } from '../../src/components/TroopsAtGoogle'
import { render } from "react-testing-library"

describe('<TroopsAtGoogle />', () => {
  let wrapper
  const data = {
    file: {
      childImageSharp: {
        fluid: {
          aspectRatio: 600,
          src: 'some-source-here',
          srcSet: 'some-source-here',
          sizes: '',
          base64: '',
        }
      },
    },
  }

  beforeEach(() => wrapper = render(<TroopsAtGoogleImage data={data}/>))

  test('should render correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
