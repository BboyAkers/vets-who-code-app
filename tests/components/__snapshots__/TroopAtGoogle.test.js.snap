// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TroopsAtGoogle /> should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class=" gatsby-image-wrapper"
        style="position: relative; overflow: hidden;"
      >
        <div
          style="width: 100%; padding-bottom: 0.16666666666666666%;"
        />
        
        <picture>
          <source
            sizes=""
            srcset="some-source-here"
          />
          <img
            alt=""
            src="some-source-here"
            style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; opacity: 1;"
          />
        </picture>
        <noscript>
          &lt;picture&gt;&lt;source srcSet="some-source-here" /&gt;&lt;img src="some-source-here" alt="" style="position:absolute;top:0;left:0;transition:opacity 0.5s;transition-delay:0.5s;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center"/&gt;&lt;/picture&gt;
        </noscript>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class=" gatsby-image-wrapper"
      style="position: relative; overflow: hidden;"
    >
      <div
        style="width: 100%; padding-bottom: 0.16666666666666666%;"
      />
      
      <picture>
        <source
          sizes=""
          srcset="some-source-here"
        />
        <img
          alt=""
          src="some-source-here"
          style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; opacity: 1;"
        />
      </picture>
      <noscript>
        &lt;picture&gt;&lt;source srcSet="some-source-here" /&gt;&lt;img src="some-source-here" alt="" style="position:absolute;top:0;left:0;transition:opacity 0.5s;transition-delay:0.5s;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center"/&gt;&lt;/picture&gt;
      </noscript>
    </div>
  </div>,
  "debug": [Function],
  "getAllByAltText": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
